content/client-lib-development-guide/features.textile

```diff
@@ -1730,11 +1734,19 @@ class RealtimeChannel:
   publish(name: String?, data: Data?) => io // RTL6i
   subscribe((Message) ->) => io // RTL7a
   subscribe(String, (Message) ->) => io // RTL7b
+  subscribe(MessageFilterObject, (Message) ->) // RTL22
   unsubscribe() // RTL8a, RTE5
   unsubscribe((Message) ->) // RTL8a
   unsubscribe(String, (Message) ->) // RTL8a
+  unsubscribe(MessageFilterObject, (Message) ->) // RTL22
   setOptions(options: ChannelOptions) => io // RTL16
 
+class MessageFilterObject:
+  hasRef: bool // RTL22b
+  refId: string // RTL22a
```

Comment:
Given we have called it `reference` in the extras rather than `ref` (which was my original suggestion), perhaps this should be `referenceId` and `referenceType`?

Reply:
I think `referenceId`/`referenceType`/`hasReference` would be needlessly verbose, I don't think you could confuse ref with any other property in this context

Reply:
Ok thanks, so do you know why we didn't use the same logic for the extras field, and used `extras.ref`? Just seems a bit inconsistent

Reply:
I think @Jmgr made the call on that change

Reply:
I can't remember the whole discussion, but I think we had `refs` at some point and then we decided to have only one, so I just called it reference. Anyway, I'm fine with renaming it `ref` if it's more coherent.

Reply:
I thought we had settled on `id`, I will change it to `timeserial`

Reply:
[Done](https://github.com/ably/docs/pull/1444/commits/82364b3dcf09bb0d93e3664687ba20ecc4079a28)